packages/api/src/external/carequality/command/cq-directory/create-or-update-cq-organization.ts

```diff
@@ -1,63 +1,104 @@
-  const orgExists = await doesOrganizationExistInCQ(org.oid);
+  const orgExists = await doesOrganizationExistInCQ(cq, cxId, org.oid);
   if (orgExists) {
-    return updateCQOrganization(cq, org);
+    return updateCQOrganization(cq, cxId, org);
   }
-  return registerOrganization(cq, org);
+  return registerOrganization(cq, cxId, org);
 }
 
-export async function doesOrganizationExistInCQ(oid: string): Promise<boolean> {
-  if (!cq) throw new Error("Carequality API not initialized");
-  const org = await cq.listOrganizations({ count: 1, oid });
-  if (org.length > 0) {
-    return true;
+async function doesOrganizationExistInCQ(
+  cq: CarequalityManagementAPI,
+  cxId: string,
+  oid: string
+): Promise<boolean | undefined> {
+  const { log } = out(`CQ get (Organization) - CQ Org OID ${oid}`);
+  if (!(await isCQDirectEnabledForCx(cxId))) {
+    log(`CQ disabled for cx ${cxId}, skipping...`);
+    return undefined;
+  }
+  try {
+    const resp = await cq.listOrganizations({ count: 1, oid });
+    return resp.length > 0;
+  } catch (error) {
+    const msg = `Failure while getting Org @ CQ`;
+    log(`${msg}. Org OID: ${oid}. Cause: ${errorToString(error)}`);
+    capture.error(msg, {
+      extra: {
+        orgOid: oid,
+        context: `cq.org.get`,
+        error,
+      },
+    });
+    throw error;
   }
-  return false;
-}
-
-export async function getCqOrganization(oid: string): Promise<Organization | undefined> {
-  if (!cq) throw new Error("Carequality API not initialized");
-  const organizations = await cq.listOrganizations({ count: 1, oid });
-  return organizations[0];
 }
 
 async function updateCQOrganization(
   cq: CarequalityManagementAPI,
+  cxId: string,
   cqOrg: CQOrganization
-): Promise<string> {
-  console.log(`Updating organization in the CQ Directory with OID: ${cqOrg.oid}...`);
+): Promise<string | undefined> {
+  const { log } = out(`CQ update (Organization) - CQ Org OID ${cqOrg.oid}`);
+  if (!(await isCQDirectEnabledForCx(cxId))) {
+    log(`CQ disabled for cx ${cxId}, skipping...`);
+    return undefined;
+  }
+
   try {
     return await cq.updateOrganization(cqOrg.getXmlString(), cqOrg.oid);
   } catch (error) {
-    console.log(
-      `Failed to update organization in the CQ Directory. Cause: ${errorToString(error)}`
-    );
+    const msg = `Failure while updating org @ CQ`;
+    log(`${msg}. Org OID: ${cqOrg.oid}. Cause: ${errorToString(error)}`);
+    capture.error(msg, {
+      extra: {
+        orgOid: cqOrg.oid,
+        context: `cq.org.update`,
+        error,
+      },
+    });
     throw error;
   }
 }
 
 async function registerOrganization(
   cq: CarequalityManagementAPI,
+  cxId: string,
   cqOrg: CQOrganization
-): Promise<string> {
+): Promise<string | undefined> {
+  const { log } = out(`CQ register (Organization) - CQ Org OID ${cqOrg.oid}`);
+  if (!(await isCQDirectEnabledForCx(cxId))) {
+    log(`CQ disabled for cx ${cxId}, skipping...`);
+    return undefined;
+  }
```

Comment:
ditto: dont think this check it necessary

Reply:
ditto above.